ClipboardAPI/clipboard-change-event-explainer.md

```diff
@@ -0,0 +1,221 @@
+
+#### 5.2.2 Approach 2 - No focus requirement
+
+##### Pros:
+1. Opens possibility for more user scenarios - e.g. For example, a web app can make network calls in the background when a specific clipboard item is updated - before pasting an image, a web app might want to check the image for embedded malicious content using a remote service. If the app can be notified about a clipboard change in background, it can trigger the necessary network operations while the user is in another page or application. When the user returns, the web app is ready with the results, reducing the wait time.
+
+##### Cons:
+1. Might be open to misuse - a web app will be able to monitor the clipboard even when the user is interacting with other applications / pages.
+2. Not much useful unless the page focus requirement is removed from the async read/write clipboard API.
+3. Could lead to higher resource consumption due to continuous monitoring. E.g. if a large number of pages are listening to the clipboardchange event, when the clipboard is changed, then all the pages will receive the event at once, which might put load on the system.
+
+#### 5.2.3 Conclusion
+We favour Approach 1 - Page required to be in focus to receive event, since this approach reduces the possibility of misusing the clipboard change event in privacy related attacks.
+
+### 5.3 Event details 
+Since the clipboardchange event is not triggered by a user action and the event is not associated to any DOM element, hence this event doesn't bubbles and is not cancellable.
+
+There are two ways to get the changed clipboard data within the event handler:
+
+1. Async Clipboard API - The [read](https://w3c.github.io/clipboard-apis/#dom-clipboard-read) or [readText](https://w3c.github.io/clipboard-apis/#dom-clipboard-readtext) methods of the [Async clipboard API](https://w3c.github.io/clipboard-apis/#async-clipboard-api) can be used to get the current contents of the system clipboard.
+
+2. DataTransfer API - The clipboardchange event is a [ClipboardEvent](https://www.w3.org/TR/clipboard-apis/#clipboard-event-interfaces) that includes a [DataTransfer](https://html.spec.whatwg.org/multipage/dnd.html#datatransfer) object. This keeps the interface of this event consistent with other clipboard related events like [cut](https://w3c.github.io/clipboard-apis/#clipboard-event-cut), [copy](https://w3c.github.io/clipboard-apis/#clipboard-event-copy) or [paste](https://w3c.github.io/clipboard-apis/#clipboard-event-paste) events. The [getData](https://html.spec.whatwg.org/multipage/dnd.html#dom-datatransfer-getdata) method of DataTransfer interface can be used to retrieve the clipboard contents of a specific format. However, it is recommended to use the Async Clipboard API because it is asynchronous and prevents the page from freezing when reading large clipboard contents.
```

Comment:
"However, it is recommended to use the Async Clipboard API because it is asynchronous and prevents the page from freezing when reading large clipboard contents." -> I think this is just a general "pitch" to use the async clipboard API instead of the DataTransfer API. Right? While that may be true, if that is tangential to the `clipboardchange` event, I would recommend excluding it from the document. Otherwise, it can create the impression that `clipboardchange` event does work as well with the DataTransfer API as the async clipboard API, which I don't think is what we are going for.